clients/src/test/java/org/apache/kafka/common/metrics/MetricsTest.java

```diff
@@ -475,28 +464,13 @@ public void testPercentiles() {
         Metric p50 = this.metrics.metrics().get(metrics.metricName("test.p50", "grp1"));
         Metric p75 = this.metrics.metrics().get(metrics.metricName("test.p75", "grp1"));
 
-        // record two windows worth of sequential values
-        for (int i = 0; i < buckets; i++)
+        // record 1-100 sequential values
+        for (int i = 0; i < buckets; i++) {
             sensor.record(i);
-
-        assertEquals(25, (Double) p25.metricValue(), 1.0);
-        assertEquals(50, (Double) p50.metricValue(), 1.0);
-        assertEquals(75, (Double) p75.metricValue(), 1.0);
-
-        for (int i = 0; i < buckets; i++)
```

Comment:
Why do we remove this code? Does it break the test?

Reply:
Yes, it breaks the test. It relies on the knowledge of internal samples/windows implementation and effectively tests windows expiration (`SampledStat`, `SampledStatTest`) instead of testing `Percentiles`. Another option was to adapt the test for recent changes in windows expiration, but it seems better to avoid using such knowledge about internals at all. What do you think?